code/modules/atmospherics/machinery/pipes/layermanifold.dm

```diff
@@ -0,0 +1,133 @@
+	front_nodes = null
+	back_nodes = null
+	nodes = list()
+	for(var/obj/machinery/atmospherics/A in needs_nullifying)
+		A.disconnect(src)
+		A.build_network()
+
+/obj/machinery/atmospherics/pipe/layer_manifold/proc/get_all_connected_nodes()
+	var/list/obj/machinery/atmospherics/all_connected = list()
+	for(var/obj/machinery/atmospherics/I in front_nodes)
+		all_connected[I] = I
+	for(var/obj/machinery/atmospherics/I in back_nodes)
+		all_connected[I] = I
+	for(var/obj/machinery/atmospherics/I in nodes)
+		all_connected[I] = I
+	var/list/obj/machinery/atmospherics/returnlist = list()
+	for(var/obj/machinery/atmospherics/A in all_connected)
+		returnlist += all_connected[A]
+	return returnlist
+
+/obj/machinery/atmospherics/pipe/layer_manifold/update_icon()	//HEAVILY WIP FOR UPDATE ICONS!!
+	layer = (initial(layer) + (PIPING_LAYER_MAX * PIPING_LAYER_LCHANGE))	//This is above everything else.
+	var/invis = invisibility ? "-f" : ""
+	icon_state = "[initial(icon_state)][invis]"
+	cut_overlays()
+	for(var/obj/machinery/atmospherics/A in front_nodes)
+		add_attached_image(A)
+	for(var/obj/machinery/atmospherics/A in back_nodes)
+		add_attached_image(A)
+
+/obj/machinery/atmospherics/pipe/layer_manifold/proc/add_attached_image(obj/machinery/atmospherics/A)
+	var/invis = A.invisibility ? "-f" : ""
+	if(istype(A, /obj/machinery/atmospherics/pipe/layer_manifold))
+		for(var/i = PIPING_LAYER_MIN, i <= PIPING_LAYER_MAX, i++)
+			var/image/I = getpipeimage('icons/obj/atmospherics/pipes/manifold.dmi', "manifold_full_long[invis]", get_dir(src, A))
+			I.pixel_x = (i - PIPING_LAYER_DEFAULT) * PIPING_LAYER_P_X
+			I.pixel_y = (i - PIPING_LAYER_DEFAULT) * PIPING_LAYER_P_Y
+			add_overlay(I)
+	else
+		var/image/I = getpipeimage('icons/obj/atmospherics/pipes/manifold.dmi', "manifold_full_long[invis]", get_dir(src, A))
+		I.pixel_x = A.pixel_x
+		I.pixel_y = A.pixel_y
+		add_overlay(I)
+
+/obj/machinery/atmospherics/pipe/layer_manifold/SetInitDirections()
+	switch(dir)
+		if(NORTH || SOUTH)
+			initialize_directions = NORTH|SOUTH
+		if(EAST || WEST)
+			initialize_directions = EAST|WEST
+
+/obj/machinery/atmospherics/pipe/layer_manifold/isConnectable(obj/machinery/atmospherics/target, given_layer)
+	if(!given_layer)
+		return TRUE
+	. = ..()
+
+/obj/machinery/atmospherics/pipe/layer_manifold/proc/findAllConnections()
+	front_nodes = list()
+	back_nodes = list()
+	var/list/new_nodes = list()
+	for(var/iter = PIPING_LAYER_MIN; iter <= PIPING_LAYER_MAX; iter += 1)
+		var/obj/machinery/atmospherics/foundfront = findConnecting(dir, iter)
+		var/obj/machinery/atmospherics/foundback = findConnecting(turn(dir, 180), iter)
+		front_nodes += foundfront
+		back_nodes += foundback
+		if(foundfront && !QDELETED(foundfront))
+			new_nodes += foundfront
+		if(foundback && !QDELETED(foundback))
+			new_nodes += foundback
+	update_icon()
+	return new_nodes
+
+/obj/machinery/atmospherics/pipe/layer_manifold/atmosinit()
+	findAllConnections()

```

Comment:
no parent call is intentional, ya?

Reply:
intentional parent call breaks it for whatever reason because i'm using a custom proc to check connections

Reply:
:+1: